support GHC 9.4 (Word# -> Word64#)

raehik opened the pull request:
GHC 9.4 switches `Word64` internal representation to `Word64#` from
`Word#`. Similar changes were made to various "Word64" primitives (which
actually worked on `Word#`. The fix is simply to apply the same wrapping
as is done for the other fixed-width primitive integers.

Addresses #3

raehik commented:
This builds & passes tests on at least 8.0.2 and 9.4.1 . Trivial change, more so than I expected.

fumieval reviewed (changes requested):
Thanks, but there are build errors on GHC 9.2

raehik commented:
Oops. `wordToWord64#` and its inverse are new to GHC 9.4, so I define them with `id` in some slightly weirder CPP.

raehik committed:
support GHC 9.4 (Word# -> Word64#)

GHC 9.4 switches `Word64` internal representation to `Word64#` from
`Word#`. Similar changes were made to various "Word64" primitives (which
actually worked on `Word#`. The fix is simply to apply the same wrapping
as is done for the other fixed-width primitive integers.

fumieval commented:
LGTM
